modelforge/potential/aimnet2.py

```diff
@@ -168,54 +169,445 @@ def compute_properties(
+            The cutoff function applied to the radial symmetry functions.
+        r_ij : torch.Tensor
+            The displacement vectors between atom pairs.
+        partial_point_charges : Optional[torch.Tensor], optional
+            The partial point charges for atoms, by default None.
+
+        Returns
+        -------
+        torch.Tensor, torch.Tensor
+            Updated atomic embeddings and partial charges.
+        """
+        combined_message = self.message_module(
+            atomic_embedding, pairlist, f_ij_cutoff, r_ij, partial_point_charges
+        )
+
+        delta_a = self.delta_a_mlp(combined_message)
+        delta_q = self.delta_q_mlp(combined_message)
+
+        return delta_a, delta_q
+
+
+class AIMNet2Representation(nn.Module):
     def __init__(
         self,
         radial_cutoff: unit.Quantity,
         number_of_radial_basis_functions: int,
+        featurization_config: Dict[str, Union[List[str], int]],
     ):
         """
-        Initialize the SchNet representation layer.
+        Initialize the AIMNet2 representation layer.
 
         Parameters
         ----------
-        Radial Basis Function Module
+        radial_cutoff : openff.units.unit.Quantity
+            The cutoff distance for the radial symmetry function.
+        number_of_radial_basis_functions : int
+            Number of radial basis functions to use.
+        featurization_config : Dict[str, Union[List[str], int]]
+            Configuration for the featurization process.
         """
         super().__init__()
 
         self.radial_symmetry_function_module = self._setup_radial_symmetry_functions(
             radial_cutoff, number_of_radial_basis_functions
         )
-        # cutoff
-        from modelforge.potential import CosineCutoff
+        # Initialize cutoff module
+        from modelforge.potential import CosineAttenuationFunction, FeaturizeInput
 
-        self.cutoff_module = CosineCutoff(radial_cutoff)
+        self.featurize_input = FeaturizeInput(featurization_config)
+        self.cutoff_module = CosineAttenuationFunction(radial_cutoff)
 
     def _setup_radial_symmetry_functions(
         self, radial_cutoff: unit.Quantity, number_of_radial_basis_functions: int
     ):
-        # NOTE: for now we are using SchNET radial symmetry function, this
-        # should be replaced with the AIMNet2 radial symmetry function
-        from .utils import SchnetRadialSymmetryFunction
+        from .utils import SchnetRadialBasisFunction
 
-        radial_symmetry_function = SchnetRadialSymmetryFunction(
+        radial_symmetry_function = SchnetRadialBasisFunction(
             number_of_radial_basis_functions=number_of_radial_basis_functions,
             max_distance=radial_cutoff,
             dtype=torch.float32,
         )
         return radial_symmetry_function
 
-    def forward(self, d_ij: torch.Tensor) -> Dict[str, torch.Tensor]:
+    def forward(self, data: Type[AIMNet2NeuralNetworkData]) -> Dict[str, torch.Tensor]:
```

Comment:
I think we should drop the `Type[...]` here. We aren't expecting the class itself as input, but rather an instance of `AIMNet2NeuralNetworkData`.